public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -488,6 +489,27 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
     // Yes this is called ever render for a function that is triggered on every mouse move
     this.eventFilter.onlyLocal = dashboard.graphTooltip === 0;
 
+    if (PanelComponent.name === 'TimeSeriesPanel' && data.series.every((df) => df.meta?.type === 'timeseries-long')) {
```

Comment:
for sure not the right place for this....   if just TimeSeriesPanel.... maybe:

https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/utils.ts#L21

or perhaps in the `prepareTimeSeries(...)` transformer?

Reply:
The `prepareTimeSeries(...)` transformer works too! It was suggested by @leeoniya that I use `partitionByValues(...)` but I am open to both suggestions. @ryantxu and @leeoniya how much different are these? Partition by Values creates a wide dataframe and Prepare Time Series allows you to choose wide or multi. According to the dataplane docs, converting TimeSeriesLong to wide modifies the data but converting to multi does not. I'm leaning towards Prepate Time Series now. @kylebrandt what do you think?

Reply:
I got some interesting side effects when I did the transformation in the `prepareGraphableFields` function
<img width="1304" alt="Screen Shot 2023-02-02 at 6 22 36 PM" src="https://user-images.githubusercontent.com/25674746/216473506-5f9ba56c-6bd3-47ea-a73a-b538cfe3435d.png">

Reply:
I think wide is best for this case. It is more performant with the time series panel as I understand it. Additionally because JavaScript provides undefined as a value for numbers (in addition to null), it doesn't have the same lossy data conversion properties that the backend does here.

Reply:
@ryantxu I tried implementing Prepare Time Series with `toTimeSeriesMulti` and with the whole transformer. Both gave me the same results, grey liners and undefined y axis ticks.

Can you help me out with what I am missing?

```
  if (data.series.every((df) => df.meta?.type === 'timeseries-long')) {
    data.series = toTimeSeriesMulti(data.series)
    
    // const options = { format: timeSeriesFormat.TimeSeriesMulti }

    // const ctx = {
    //   interpolate: (value: string) => value,
    // }

    // data.series = prepareTimeSeriesTransformer.transformer(options, ctx)(data.series)
  }
```